### ISSUE TITLE
Guest cart API ignoring cartId in url for some methods

### ISSUE THREAD

**Issue by midlan** [magento/magento2#14086]
### Preconditions
1. Magento 2.2.3

### Steps to reproduce
I am following the [API documentation](http://devdocs.magento.com/swagger/index.html) to create guest cart and add item to it:
1.  `POST https://yourdomain/rest/default/V1/guest-carts` (with empty body) creates cart for you and return **cartId** in body
2.  `GET https://yourdomain/rest/default/V1/guest-carts/`**`:cartId`** to check you cart is created
3. `POST https://yourdomain/rest/default/V1/guest-carts/`**`:cartId`**`/items` with body

```
{
  "cartItem": {
    "sku": "WS12-M-Orange",
    "qty": 3
  }
}
```
to add item to your cart, but error is returned instead of adding cart to item

### Expected result
In step 3, I was expecting the added item to be returned. Instead of that, error is returned.

### Actual result
```
{
    "message": "No such entity with %fieldName = %fieldValue",
    "parameters": {
        "fieldName": "cartId",
        "fieldValue": null
    },
    "trace": "..."
}
```

### Further research
I discovered: If you send the **cartId** in `quote_id` field in body at step 3, the method would work. So the body would be:
```
{
  "cartItem": {
    "sku": "WS12-M-Orange",
    "qty": 3,
    "quote_id": ":cartId"
  }
}
```
And the results is cart item, no error. The **cartId** in url is completly ignored. If you use send request with some dummy string instead of cartId in URL (with the body with quote_id field), it actually works.

So request like
`POST https://yourdomain/rest/default/V1/guest-carts/some_dummy_string_insetead_of_cartId/items`
works. But it should not! And the mandatory quote_id in body is also wrong I think. The only cartId needed to add the item to the cart should be the one from URL (which is not used at all at the moment).

**The same problem is present on other methods in guest-cart.**

**Comment by magento-engcom-team** [comment]
@midlan, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by elzekool** [comment]
I've created a PR for this issue. Investigating this a little further I found a related issue for the normal cart API. In the case /V1/carts/mine/items is used it allows adding/editing items to a quote different then the actual customer quote.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-30200` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @midlan. Thank you for your report. 
The issue has been fixed in magento/magento2#27172 by @engcom-Charlie in 2.4-develop branch
Related commit(s):
 - [2bdcc11de0bd8d127a3115383ea960577c840920](https://github.com/magento/magento2/commit/2bdcc11de0bd8d127a3115383ea960577c840920)
 - [23f98498398e58b755ddffbe65601633402ea0af](https://github.com/magento/magento2/commit/23f98498398e58b755ddffbe65601633402ea0af)
 - [e3c2951cb7fe8c25afc3ad7ee8a8cda8c64422d0](https://github.com/magento/magento2/commit/e3c2951cb7fe8c25afc3ad7ee8a8cda8c64422d0)

The fix will be available with the upcoming 2.4.0 release.
